source/dub/dub.d

```diff
@@ -1380,7 +1380,11 @@ class Dub {
 		import std.process : environment;
 		import std.range : front;
 
-		m_defaultCompiler = m_config.defaultCompiler.expandTilde;
+		// Env takes precedence
+		if (auto envCompiler = environment.get("DC"))
+			m_defaultCompiler = envCompiler;
+		else
+			m_defaultCompiler = m_config.defaultCompiler.expandTilde;
```

Comment:
Does `env` take precedence, or is it just looked up first ? IIUC `--compiler` takes precedence over everything else ?

Reply:
it takes precedence over the m_config. But --compiler overrides everything, yes.